frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/storage-tab-state-update.ts

```diff
@@ -16,13 +16,24 @@ import { DiskWrapper } from '../../../../k8s/wrapper/vm/disk-wrapper';
 
 export const prefillInitialDiskUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   const state = getState();
-  if (!hasVmSettingsChanged(prevState, state, id, VMSettingsField.PROVISION_SOURCE_TYPE)) {
+  const baseDiskImageChanged =
+    (hasVmSettingsChanged(prevState, state, id, VMSettingsField.OPERATING_SYSTEM) ||
+      hasVmSettingsChanged(prevState, state, id, VMSettingsField.FLAVOR) ||
+      hasVmSettingsChanged(prevState, state, id, VMSettingsField.WORKLOAD_PROFILE)) &&
+    iGetVmSettingValue(state, id, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE);
```

Comment:
why not checking this one as well to reset the disk to null if needed?

Reply:
done :balloon: